site/content/docs/getting-further/automation/target-scanning-issues.md

```diff
@@ -0,0 +1,187 @@
+---
+title: "Automation Guide - Target Scanning Issues"
+tags: 
+- automation
+- guide
+type: page
+---
+
+It is not unusual for target systems to struggle or even fail when being scanned by ZAP.
+
+This page explains what can go wrong, how to detect these problems and what can be done about them.
+
+### Cannot Connect
+
+If ZAP cannot connect to the target app then it will typically fail very quickly.
+Solving connection problems will depend on the underlying cause, which ZAP will not be able to detect.
+
+To confirm whether this is just a ZAP problem or something unrelated to ZAP then use another tool like [curl](https://curl.se/) to access the target from 
+exactly the same environment that ZAP is running in.
+
+#### Detection
+In this case detection is typically straightforward - exploring will fail.
+
+#### Target is Down
+This is nice and easy - start the target app!
+
+#### ZAP blocked by Firewall
+In this case the firewall needs to be configured to give ZAP access.
+
+This can be done via:
+* IP address range
+* A [Custom HTTP header](#custom-http-header)
+
+#### No Network Route
+You’ll need to sort out your networking.
+
+### Connection Fails Mid Scan
+If ZAP cannot connect to the target “mid scan” then the scan could carry on for a significant length of time.
+For example, if ZAP has completed the exploration phase then it will already know about all of the target endpoints.
+It will currently carry on trying to attack them even if all of the connections fail.
+
+#### Detection
+To detect connection failures monitor the global statistic:
+* `stats.network.send.failure`
+
+This is a count of all of the times ZAP has failed to send an HTTP message.
+Depending on your environment you may get some occasional failures, so it might not be a good idea to fail if you just see a few failures. If the connection to the target is lost during a scan then the number of failures will increase rapidly.
+
+There is a corresponding statistic which counts the successful sends:
+* `stats.network.send.success`
+
+Note that these are new statistics which are currently only available in the Dev and Live builds!
+
+#### Target Crashes
+This is not uncommon as ZAP can put a very heavy load on the target app.
+
+Ideally you should fix the target so that it doesn't crash!
+As a work around you can use ZAP [Rate Limiting](#rate-limiting).
+
+#### Firewall Blocks ZAP
+See [ZAP blocked by Firewall](#zap-blocked-by-firewall) and [Rate Limiting](#rate-limiting).
```

Comment:
Combine this with FW bit earlier?

Reply:
Maybe I'm not keeping track of header levels

Reply:
its under a different heading